allennlp/interpret/attack/hotflip.py

```diff
@@ -0,0 +1,184 @@
+from typing import Dict, List, Set
+import numpy
+import torch
+from allennlp.interpret.attack import Attacker
+from allennlp.common.util import JsonDict, sanitize 
+from allennlp.data import Instance
+from allennlp.predictors.predictor import Predictor
+from allennlp.data.fields.field import DataArray, Field
+from allennlp.data.fields import IndexField
+from allennlp.modules.text_field_embedders.text_field_embedder import TextFieldEmbedder
+from allennlp.modules.text_field_embedders import BasicTextFieldEmbedder
+from allennlp.modules.token_embedders import Embedding
+from allennlp.data.tokenizers import Token
+from collections import defaultdict
+
+@Attacker.register('hotflip')
+class Hotflip(Attacker):
+    def __init__(self, predictor):
+        """
+        Runs the HotFlip style attack at the word-level https://arxiv.org/abs/1712.06751.
+        We use the first-order taylor approximation described in https://arxiv.org/abs/1903.06620,
+        in the function first_order_taylor().
+        """
+        super().__init__(predictor)                                          
+        self.vocab = self.predictor._model.vocab
+        """ For HotFlip, we need a word embedding matrix to search over. The below is neccessary for
```

Comment:
Hmm, actually, this whole block should be pulled out into its own method, like `_construct_embedding_matrix`, and then this could stay a docstring on that method (with maybe also a comment on the class docstring that constructing the object is expensive).  And then you just have a line in the constructor like `self.token_embedding = self._construct_embedding_matrix()`.